### ISSUE TITLE
Anyway to obtain the local filepath from <input type="file".../>?

### ISSUE THREAD

**Issue by ffreemt** [alifcommunity/webui#46]
Suppose I have `<input id="filepath" type="file" accept=".txt" />` in html, how do I get the local filepath of the uploaded file? 

All I got is `C:\fakepath\tmp.txt`. I need to process the local file after a user select a file.

Thanks for the nice package. I use python and quite enjoy it.

**Comment by hassandraga** [comment]
For security reasons, all web browsers hide the real absolute file path. But you can easily use the native OS APIs instead. For example, you call your Python function that opens a file dialog and returns the absolute path to the UI.

```
import win32gui, win32con, os

filter='Python Scripts\0*.py;*.pyw;*.pys\0Text files\0*.txt\0'
customfilter='Other file types\0*.*\0'
fname, customfilter, flags=win32gui.GetOpenFileNameW(
    InitialDir=os.environ['temp'],
    Flags=win32con.OFN_ALLOWMULTISELECT|win32con.OFN_EXPLORER,
    File='somefilename', DefExt='py',
    Title='GetOpenFileNameW',
    Filter=filter,
    CustomFilter=customfilter,
    FilterIndex=0)

print 'open file names:', repr(fname)
print 'filter used:', repr(customfilter)
print 'Flags:', flags
for k,v in win32con.__dict__.items():
    if k.startswith('OFN_') and flags & v:
        print '\t'+k
```

Or use a third-party module

```
import easygui
path = easygui.fileopenbox()
```

**Comment by hassandraga** [comment]
JavaScript

```
var result = webui_fn('MyPythonFunction');
alert('Path is: ' + result);
```

**Comment by ffreemt** [comment]
Thanks very much for the reply and for the detailed alternatives.

I  am aware of the security reasons for all web browsers to hide the real absolute file path.

But since this webui package is intended for processing local content, I wonder whether there is a way to directly access the real absolute file path like `electron.js` does.

**Comment by hassandraga** [comment]
`Electron` uses an entire embedded web browser (_Chromium_), that's why the final result is +100 MB. While WebUI uses only the already installed real web browser. Perhaps, probably we should add an API for opening a file, something like `var path = webui_fopen('*.txt');`

**Comment by ffreemt** [comment]
I see. I look forward to the possible new api. In the meantime, I use a somewhat alright workwround: limit the directories to load a file, for example in home directory and current work directory, the filename itselft can be obtained from `<input type="file"... >` in html (e.g., `C:\fakepath\tmp.txt`).

**Comment by gaycomputers** [comment]
Hey, if you aren't using the browsers file selector I'm not sure this workaround is guaranteed to work on some platforms. For example I am using golang and I'm not sure how to guarantee a file selector will actually open on linux; they could be using firefox on i3 with no installed file browser for all I know... I am new to the project and notice the api just switched, did the file thing ever get explicitly worked out?

**Comment by gaycomputers** [comment]
It sounds stupid when I think about it too long and I'm not a web dev but could you stream the data over the browser like uploading/downloading a file from a website? I have no idea how to code this on webui

**Comment by AlbertShown** [comment]
I already saw the file handling using HTML5/JS in the [text editor](https://github.com/webui-dev/webui/tree/main/examples/C/text-editor) example.

Another way to do this is by making the backend use OS native APIs to open the file selector, then send the file content to UI using [webui_send_raw()](https://github.com/webui-dev/webui/blob/4c3932ee0084aa0dd20f2446a0f5337f77184219/include/webui.h#L377C7-L377C7).

![ScreenShot](https://user-images.githubusercontent.com/34311583/255719171-c1ccf29c-806a-4742-bfd7-a3dc30cab70b.png)

**Comment by gaycomputers** [comment]
> I already saw the file handling using HTML5/JS in the [text editor](https://github.com/webui-dev/webui/tree/main/examples/C/text-editor) example.
> 
> Another way to do this is by making the backend use OS native APIs to open the file selector, then send the file content to UI using [webui_send_raw()](https://github.com/webui-dev/webui/blob/4c3932ee0084aa0dd20f2446a0f5337f77184219/include/webui.h#L377C7-L377C7).
> 
> ![ScreenShot](https://user-images.githubusercontent.com/34311583/255719171-c1ccf29c-806a-4742-bfd7-a3dc30cab70b.png)

I think that using the OS native APIs was hassandraga's first suggestion, but I'm planning to support platforms without native file browsers. I'm interested in this text editor example I haven't seen before. I'm using golang currently, I was able to modify the code of the text editor example to transfer the data of the file. This actually works for me at the moment, though it's kind of a stupid workaround for what I'm doing (loading/saving configs/databases). Next I have to figure out how to save the config to a new file (using the web file browser).

**Comment by hassandraga** [comment]
If your data is bigger than ~250Kb it won't work actually using WebUI v2.4.0. But will be fixed soon in 2.4.1.
